WordPress/gutenberg
# Allow creating custom block templates in classic themes

The idea of this PR is to allow creating custom block templates (starting blank) for a given page in classic themes. It's a way for classic themes to enable some FSE features without any change.

**How to test it**

It's very rough at the moment, more a POC, but here's how  you do  it:

 - Create  a new page
 - Save it
 - you'll notice on the Document sidebar, a "Create custom template" link
 - Once you click that link a custom block template is created and attached to the page.
 - By default that template Is composed by just two blocks (post title and post content) but you can edit as you wish
 - Apply the changes and publish

Now, on the frontend, your page is using its own custom template and not the regular `page.php` or `singular.php` that comes from the theme itself.

**Notes**

This PR can be seen as a migration strategy for classic themes, it highlights a few things in terms of code/architecture:

 - The is_fse_theme checks become less important as this blurs the line between classic and FSE themes, the whole FSE infrastructure (template post type, template resolution, template API) need to be available for classic themes too.
 - In terms of design it's very rough at the moment, we need to improve that. I'm looking at you designers :P @jasmussen @jameskoster 
 - All FSE blocks become available for classic themes with this PR, this means if we ship this to Core, we need to make some good decisions about which FSE blocks are considered stable.
 - Right  now, it's enable for all classic themes by default, we might want to introduce a theme support flag for it to make it opt-in.

**Todo**

 - [x] Allow setting titles when creating custom templates.
 - [x] Improve the design a bit.
 - [x] Write an e2e test.
 - [x] Add some documentation.
 - [x] Theme support flag.
